### DISCUSSION TITLE
PP values for global leaderboard scores are either incorrect or displayed as 0

### DISCUSSION THREAD

**Discussion by Flamiii** [ppy/osu#19222] [Q&A]
Map: https://osu.ppy.sh/beatmapsets/1759729#osu/3601629

When looking at scores on the global leaderboard, the pp seems to be calculated correctly in the performance breakdown but will not be properly displayed under the score.

For example in this score, the actual pp that should be displayed is 461pp since this is a DT play that is only 1.1x, but it is treated as a normal DT play that is 1.5x instead, which displays a far higher pp value of 895pp which isn't accurate.

![osu_2022-07-18_16-09-21](https://user-images.githubusercontent.com/74088315/179617775-5591780d-736b-44e6-8ef8-6e8ea15af13c.png)

Scores with certain mods enabled such as Difficulty Adjust, Relax, Random, Autopilot, Target, and Magnetised also always seem to show 0pp despite the performance breakdown seeming to accurately calculate the pp value based on the mod. There are probably other mods that this occurs with as well.

![osu_2022-07-18_16-13-14](https://user-images.githubusercontent.com/74088315/179618476-09551588-62a7-4544-b27d-770a0ca27869.png)

There are also scores that show a pp value that seems somewhat correct, but the performance breakdown will display a different pp value entirely.

![osu_2022-07-18_16-20-21](https://user-images.githubusercontent.com/74088315/179619452-8c854c98-529a-40a0-8c8b-4c7becd59f32.png)

I'm not really sure if the pp value displayed in the performance breakdown is 100% accurate, but I still think it's kind of confusing to have 2 separate pp values displayed under the same score. The pp values shown on local scores always seem to match the value shown in the performance breakdown.

**Comment by smoogipoo** [comment]
I think the answer to most of these is likely going to be values coming in from online sources (i.e. osu-queue-score-statistics), and:
a) For the first screenshot, the processor uses the in-database difficulty attributes, so doesn't properly consider rates.
b) For the second screenshot, the processor prevents [certain mods from calculating](https://github.com/ppy/osu-queue-score-statistics/blob/f288bd7abbe8d29fa2fe07968b1d0ef0e21230f3/osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs#L143-L184).
c) For the third screenshot, scores _probably_ haven't been reprocessed. It will _probably_ be fixed when we do a full reprocess.

And in all cases above, the performance breakdown uses the local values.

**Reply by peppy** [reply]
One improvement is probably using the "online" pp in the performance breakdown for consistency (or showing one or the other as a tooltip for reference purposes).
